CONTRIBUTING.md

```diff
@@ -107,3 +107,54 @@ MyButton : Add unit test
 
 Files changed:
 - MyButtonTest.py
+
+
+## Code reviews and labels ##
+
+### Label guidelines ###
+For labels, our primary concern is to clearly and consistently distinguish between Issue labels and Pull Request (PR) labels. 
+
+For an up-to-date list of Gaffer's GitHub labels, see the project's [Labels](https://github.com/GafferHQ/gaffer/labels) page.
+
+We have the following general specs for labels:
+- Unfortunately, there is currently no way to maintain separate lists of labels based on their intended use. Therefore, PR labels are prefixed with `pr-`.
+- PR labels that communicate that the PR needs immediate attention have high-intensity colors.
+- PR labels that communicate that the PR should not be merged yet have medium-intensity colors.
+- Issue labels have no prefix, and have low-intensity colors.
+
+#### Issue labels ####
+Examples: `arnold` `bug`
+
+The purpose of Issue labels is to help us filter and sort our active Issues. We have chosen very subtle colors and short names for them.
+
+If you're creating an Issue, please apply as many of these labels that match. **Do not** apply these labels to PRs.
+
+#### Pull Request labels ####
+Examples: `pr-hold` `pr-revision`
+
+The purpose of PR labels is to signal which PRs can't be merged yet. A PR could be unready for merging for two reasons: the PR needs more work (`pr-revision` and `pr-docs`) or the PR must wait for something else to happen in the code base (`pr-hold` and `pr-blocked`).
+
+### Pull Request reviews ###
+In order to effectively and efficiently maintain our Pull Requests (PRs), our contributors and reviewers must follow these PR guidelines:
+
+#### When creating a PR (contributor) ####
+* Assign yourself as the PR's Assignee. The Assignee is the PR's stakeholder/owner.
+* Assign at least one contributor as a Reviewer. We will not merge any PRs that haven't been reviewed.
```

Comment:
I think this is asking too much - an outside contributor has no way of knowing who is the best person to review. We can assign reviewers ourselves as necessary. I'm also not quite sure what the rationale for assigning Assignee is - it's not something I've found very useful.

Reply:
Re outside contributors, yeah thats fair, they should leave it to a maintainer to assign the reviewer.

Re Assignee - Tom started doing it for his own, and I found it incredibly useful, so I asked everyone to follow his lead on that. It makes it very easy to scan the PR page and see who made which PR (via icons), as opposed to reading all the tiny subtext to find the author name. Its also the GitHub documented meaning of Assignee -- the person doing the work, which may not be the original PR author in the (admittedly rare) case of a takeover. I've requested they automate the assignee on PR creation and they agreed it'd be a useful feature (though didn't commit to a timeline)

Reply:
> I think this is asking too much - an outside contributor has no way of knowing who is the best person to review. We can assign reviewers ourselves as necessary.

https://github.com/GafferHQ/gaffer/pull/3266/commits/2fbddd227f035e24e9f067628b74d82ec0c8e692#diff-6a3371457528722a734f3c51d9238c13R126-R127